### ISSUE TITLE
TextToToken transformation contribution

### ISSUE THREAD

**Issue by lukzas** [Seddryck/NBi#621]
I've implemented `text-to-token` native transformation to use in my project. It enables to split string with a default `|` separator and select value of the given index, as well as custom one, passed as parameter.

### Function calls
```xml
text-to-token(index)
text-to-token(index, separator)
```

### Example
```xml
<local-variable name="Table" type="text">
  <item>dbo;Service</item>
</local-variable>
<derived-variable name="TableSchema" based-on="Table" type="text">
  <script language="native">
    text-to-token(0,;)
  </script>
</derived-variable>
<derived-variable name="TableName" based-on="Table" type="text">
  <script language="native">
    text-to-token(1,;)
  </script>
</derived-variable>
```

I've tested it in many different scenarios already as it's handier than the custom `c-sharp` transformation. In case you're interested in adding it to the codebase, let me know and I'll create PR.

**Comment by Seddryck** [comment]
Yes, surely, sounds as a great improvement! Just that I'll probably have to change the default separator to match with existing implementation of `text-to-token-count`.

**Comment by lukzas** [comment]
I might adjust it to use whitespace as default instead of the current character :wink:

**Comment by Seddryck** [comment]
Would be great. Looking forward for your contribution.

**Comment by Seddryck** [comment]
@lukzas I'd like to make a couple of updates:

- Current default separator is the space char, I'd like to change it to any whitespace such as the `text-to-token-count` function
- I also would like to align the behaviour when you've consecutive separators between the two functions
- You inherited from TextToLength but it probably means that applying the functions to `(null)` will return 0, would like to return `(null)` in place
- Currently when you're requesting a token that doesn't exist, you throw an indexoutofrange exception, I'd prefer to return `(null)` to have the same kind of behaviour than other text methods.

Is it OK for you?

**Comment by lukzas** [comment]
@Seddryck All seem valid points, especially in terms of consistency within the repository. Though the current implementation works for all my use cases and e.g. having support for whitespace other than space is totally unnecessary. Feel free to adjust it as you find most appropriate for the project 😉

**Comment by Seddryck** [comment]
Implemented and documented in [1.23.0-beta.161](https://ci.appveyor.com/project/Seddryck/nbi/builds/37254279/artifacts) or on nuget via ```Update-Package NBi.Framework -version 1.23.0-beta0161```.
